### ISSUE TITLE
NullpointerException in Topic

### ISSUE THREAD

**Issue by chrosim** [vaadin/collaboration-engine#55]
I am facing a NPE sometimes which is raised in the Topic.class
More specific inside the clearExpiredData Method the namedMapData.get(name) sometimes results in a null value.

```java
private void clearExpiredData() {
        Clock clock = collaborationEngine.getClock();
        if (lastDisconnected != null) {
            Instant now = clock.instant();
            mapExpirationTimeouts.forEach((name, timeout) -> {
                if (now.isAfter(lastDisconnected.plus(timeout))) {
                    namedMapData.get(name).clear();
                }
            });
            listExpirationTimeouts.forEach((name, timeout) -> {
                if (now.isAfter(lastDisconnected.plus(timeout))) {
                    namedListData.get(name).clear();
                }
            });
        }
        lastDisconnected = null;
    }
```

**Versions**
    - Vaadin version: 14.8.0
    - Collaboration Engine version: 3.2.2
    - Java version: 11

**Comment by Legioth** [comment]
Seems like this can happen in case a timeout has been set for a map that never has had any content. Can be tested with this variation of the regular expiration timeout test (`expirationTimeout_mapClearedAfterTimeout`):

```
@Test
public void expirationTimeout_unpopulatedMap_mapClearWithoutExplosions() {
    Duration timeout = Duration.ofMinutes(15);
    map.setExpirationTimeout(timeout);
    registration.remove();
    ce.setClock(Clock.offset(ce.getClock(), timeout.plusMinutes(1)));

    ce.openTopicConnection(context, "topic", SystemUserInfo.getInstance(), connection -> null);
    // All is fine if openTopicConnection runs successfully
}

**Comment by heruan** [comment]
@chrosim the issue has been fixed in Collaboration Engine [5.0.0.alpha2](https://github.com/vaadin/collaboration-engine/releases/tag/5.0.0.alpha2) which requires Vaadin [23.0.0.alpha3](https://github.com/vaadin/platform/releases/tag/23.0.0.alpha3).

We are backporting the fix also to Collaboration Engine 3.2 and 4.0, I'll let you know once released.
